---
title: "snRNA-sequencing of Aortic Dissection"
author: "Mark E. Pepin, MD, PhD, MS"
date: "01/09/2023"
output:
  html_document:
    code_folding: hide
    keep_md: yes
    toc: yes
    toc_float: yes
header-includes:
- \usepackage{booktabs}
- \usepackage{longtable}
- \usepackage{array}
- \usepackage{multirow}
- \usepackage[table]{xcolor}
- \usepackage{wrapfig}
- \usepackage{float}
- \usepackage{colortbl}
- \usepackage{pdflscape}
- \usepackage{tabu}
- \usepackage{threeparttable}
mainfont: Times
fontsize: 10pt
always_allow_html: yes
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include=FALSE}
library(knitr)
library(kableExtra)
opts_chunk$set(tidy.opts=list(width.cutoff=30),tidy=FALSE, warning = FALSE, message = FALSE, cache = TRUE, cache.lazy = FALSE)
options(knitr.kable.NA = '')
```

**Code Authors**: Mark E. Pepin, MD, PhD, MS **Contact**:
[pepinme\@gmail.com](mailto:mepepin@bwh.harvard.edu){.email}\
**Institution**: Brigham and Women's Hospital | Broad Institute of Harvard and MIT\
**Location**: Boston, MA

# Data Pre-Processing

```{r preprocess, echo = FALSE, message=FALSE, warning=FALSE}
library(Seurat)
# Upload data provided
hdat <- readRDS(file = "../1_Input/snRNA_Human_ControlDissection_integrated_seurat_v2.rds")
# Initialize the Seurat object with the raw (non-normalized data).
hdat[["percent.mt"]] <- PercentageFeatureSet(hdat, pattern = "^MT-")
hdat <- subset(hdat, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 5)
hdat <- NormalizeData(hdat, normalization.method = "LogNormalize", scale.factor = 10000)
hdat <- FindVariableFeatures(hdat, selection.method = "vst", nfeatures = 10000)
all.genes <- rownames(hdat)
hdat <- ScaleData(hdat, features = all.genes)
hdat[["Celltype_annotation_v2"]]$Celltype_annotation_v2 <- factor(hdat[["Celltype_annotation_v2"]]$Celltype_annotation_v2, c("VSMC1", "VSMC2", "Myofibroblast", "Fibroblast1", "Fibroblast2", "EC1", "EC2", "Macrophage", "NKT Cells", "Neuronal"))
##
TAA_list<-SplitObject(hdat, split.by = "Disease") # split dataset into a list of two seruat objects
# normalize and identify variable features for each dataset independently
TAA.list <- lapply(X = TAA_list, FUN = function(x) {
    x <- NormalizeData(x)
    x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 10000)
})
# select features that are repeatedly variable across datasets for integration
features <- SelectIntegrationFeatures(object.list = TAA.list)
# Preserve cell type markers in the anchoring dataset
VSMC1_genes<-c("PKD1", "COL6A2", "PDLIM7", "FLNA", "SMTN")
VSMC2_genes<-c("MYH11", "ACTA2", "ITGA8", "PRKG1", "CRISPLD1")
Fibroblast1_genes<-c("ABCA10", "C3", "ADGRD1", "FBLN1", "DCN")
Fibroblast2_genes<-c( "NFASC",  "SAMD5","UACA","PRSS23") #  "TEX41",
Fibromyocyte_genes<-c("DGKG", "ADAMTS1", "RGS6", "TNC", "GRIP2", "ANGPT2")
EC1_genes<-c("DIPK2B", "ARHGEF15", "STC1", "FLT1") #, "NOTCH4"
EC2_genes<-c("VWF", "BMPER", "BMX", "NOS1")
NKT_genes<-c("SKAP1", "RIPOR2", "RBPJ", "FYN", "ITGAL", "CD96")
Macrophage_genes<-c("MRC1", "LGMN", "F13A1", "RBM47")
Dendritic_genes<-c("ITGAX", "S100A9", "CSF3R", "CXCL8")
features<-union(features, union(VSMC1_genes,VSMC2_genes))
#   VSMC1_genes, union(
#     VSMC2_genes, union(
#       Fibroblast1_genes, union(
#         Fibroblast2_genes, union(
#           Fibromyocyte_genes, union(
#             EC1_genes, union(
#               EC2_genes, union(
#                 NKT_genes, union(Macrophage_genes, Dendritic_genes)
#                 )
#               )
#             )
#           )
#         )
#       )
#   )
# )
# )
# Identify "anchors" (cells that are unchanged across datasets)
TAA.anchors <- FindIntegrationAnchors(object.list = TAA.list, anchor.features = features)
# this command creates an 'integrated' data assay
TAA.combined <- IntegrateData(anchorset = TAA.anchors)
# Specify the matrix used in the downstream analysis
DefaultAssay(TAA.combined) <- "integrated"
# Run the standard workflow for visualization and clustering
TAA.combined <- ScaleData(TAA.combined, verbose = FALSE)
TAA.combined <- RunPCA(TAA.combined, npcs = 30, verbose = FALSE)
TAA.combined <- RunUMAP(TAA.combined, dims = 1:30)
TAA.combined <- RunTSNE(TAA.combined)
TAA.combined <- FindNeighbors(TAA.combined, dims = 1:30)
TAA.combined <- FindClusters(TAA.combined, resolution = 0.4)
saveRDS(TAA.combined, file = "TAA_FFPE_snRNA.rds")
#
plot1 <- VariableFeaturePlot(hdat)
top10 <- head(VariableFeatures(hdat), 10)
plot2 <- LabelPoints(plot = plot1, points = c(top10), repel = TRUE)
pdf(file = "../2_Output/Figure_1/FFPE_Variable_genes.pdf")
plot2
dev.off()
```

# Figure 1
## Clustering

```{r cluster_combined, echo = FALSE, message=FALSE, warning=FALSE}
# Clustering
umap_disease <- DimPlot(hdat, reduction = "umap", group.by = "Disease")
umap_clusters <- DimPlot(TAA.combined, reduction = "umap", label = TRUE, repel = TRUE)
pca_disease <- DimPlot(TAA.combined, reduction = "pca", group.by = "Disease")
pca_clusters <- DimPlot(TAA.combined, reduction = "pca", label = TRUE, repel = TRUE)
tsne_disease <- DimPlot(TAA.combined, reduction = "tsne", group.by = "Disease")
tsne_clusters <- DimPlot(TAA.combined, reduction = "tsne", label = TRUE, repel = TRUE)
pdf("../2_Output/Supplemental_Figures/Clustering_Comparison.pdf", width = 11, height = 8)
umap_disease+pca_disease+tsne_disease+umap_clusters+pca_clusters+tsne_clusters
dev.off()

# Add tracings around the points
library(ggplot2)
library(ggtrace)
library(ggthemes)
library(ggrepel)
library(dplyr)
# UMAP
umap_data <- as.data.frame(TAA.combined@reductions$umap@cell.embeddings) # Create a data.frame from the UMAP (to be used in ggplot)
# Disease Clustering
umap_data$Disease <- TAA.combined$Disease
umap_data$Disease <- factor(umap_data$Disease, levels = c("Control", "Dissection"))
umap_disease <- ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, fill = Disease)) +
  geom_point_trace(alpha = .25, stroke = .5, size = 1, color = "black")  +
  theme_classic() +
  theme(text = element_text(size = 14))
umap_disease
### UMAP Clustering
umap_data$Cluster <- TAA.combined$seurat_clusters # Create a variable based on UMAP clusters (Seurat)
cluster_centers <- aggregate(cbind(UMAP_1, UMAP_2) ~ Cluster, umap_data, mean) # Label Clusters
umap_clusters <- ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, fill = Cluster)) +
  geom_point_trace(alpha = .25, stroke = .5, size = 1, color = "black")  +
  theme_classic() + 
  geom_text_repel(data = cluster_centers, aes(label = Cluster), size = 6) +
  theme(legend.position = "none", text = element_text(size = 14))
umap_clusters
pdf(file = "../2_Output/Supplemental_Figures/UMAP_overlay.pdf", width = 11.5, height = 5, bg = "transparent")
umap_disease + umap_clusters
dev.off()

#Figure 1C -  Proportional Graph
library(dittoSeq)
dittoBarPlot(
    object = hdat,
    var = "Celltype_annotation_v2",
    group.by = "Disease")
dittoBarPlot(
    object = hdat,
    var = "Celltype_annotation_v2",
    group.by = "Sample_name",
    split.by = "Disease")

######################################################
############## Dimensionality
#####################################################
# to control for any confounders, we can add the 'vars.to.regress' parameter, including any variable in the metadata <- ScaleData(pbmc, vars.to.regress = "percent.mt")
TAA.combined <- RunPCA(TAA.combined, features = VariableFeatures(object = TAA.combined))
# Visualize the features/genes
VizDimLoadings(TAA.combined, dims = 1:5, reduction = "pca")
DimPlot(TAA.combined, reduction = "umap")
pdf(file = "../2_Output/Supplemental_Figures/PC_Heatmaps.pdf")
DimHeatmap(TAA.combined, dims = 1:15, cells = 500, balanced = TRUE)
dev.off()
# Determine the number of dimensions represented by the data
TAA.combined <- JackStraw(TAA.combined, num.replicate = 100)
TAA.combined <- ScoreJackStraw(TAA.combined, dims = 1:20)
pdf(file = "../2_Output/Supplemental_Figures/JackStraw.pdf")
JackStrawPlot(TAA.combined, dims = 1:20)
dev.off()
pdf(file = "../2_Output/Supplemental_Figures/ElbowPlot.pdf")
ElbowPlot(TAA.combined)
dev.off()
######################################################
############## Cluster Annotation
#####################################################
library(ggplot2)
library(dplyr)
library(Seurat)
# TAA.combined<-readRDS(file = "TAA.ALL_snRNA.rds") # contains ACTA2, normal, and aortic aneurism (Cheu et. al)
# Find all gene markers
DEGs_Clusters<-FindAllMarkers(TAA.combined, assay = "RNA")
write.csv(DEGs_Clusters, "../2_Output/Supplemental_Figures/DEGs_Clusters.csv")
#Identify Clusters corresponding with known gene markers:
pdf(file = "../2_Output/Supplemental_Figures/UMAP_split.pdf", height = 4, width = 7)
DimPlot(TAA.combined,  label = T, split.by = "Disease") + NoLegend()
dev.off()
pdf(file = "../2_Output/Supplemental_Figures/UMAP_samples.pdf", height = 4, width = 11)
DimPlot(TAA.combined,  label = T, split.by = "Sample_name") + NoLegend()
dev.off()
# Create a dot-bplot of the top 5 markers for each 
library(scCustomize)
paletteLength <- 100
myColor <- colorRampPalette(c("dodgerblue4", "white", "coral2"))(paletteLength)
top5_markers <- Extract_Top_Markers(marker_dataframe = DEGs_Clusters, num_genes = 10, named_vector = FALSE,
    make_unique = TRUE)
pdf(file = "../2_Output/Supplemental_Figures/Dot.plot_top5markers.pdf", height = 10, width = 7)
Clustered_DotPlot(seurat_object = TAA.combined, features = top5_markers, k = 10, colors_use_exp = myColor)
dev.off()
saveRDS(TAA.combined, file = "TAA_FFPE_snRNA.rds")
```

## Cell-type Annotation

```{r merged.annotation, echo = FALSE, message=FALSE, warning=FALSE}
library(Seurat)
library(ggrepel)
TAA.combined<-readRDS(file = "TAA_FFPE_snRNA.rds")
#Identify Clusters corresponding with known gene markers:
VSMC1_genes<-c("PKD1", "COL6A2", "PDLIM7", "FLNA", "SMTN")
VSMC2_genes<-c("MYH11", "ACTA2", "ITGA8", "PRKG1", "CRISPLD1")
Fibroblast1_genes<-c("ABCA10", "C3", "ADGRD1", "FBLN1", "DCN")
Fibroblast2_genes<-c("NFASC",  "SAMD5", "PRSS23") #"UACA","TEX41",
Fibromyocyte_genes<-c("ADAMTS1", "RGS6", "TNC") # , "ANGPT2", "DGKG", "GRIP2"
EC1_genes<-c("DIPK2B", "ARHGEF15", "STC1", "FLT1") # , "NOTCH4"
EC2_genes<-c("VWF", "BMPER", "BMX", "NOS1")
NKT_genes<-c("SKAP1", "RIPOR2", "ITGAL", "CD96") #  "RBPJ", "FYN",
Macrophage_genes<-c("MRC1", "LGMN", "F13A1", "RBM47") #
Dendritic_genes<-c("ITGAX", "S100A9", "CSF3R", "CXCL8") #
# Plot density function
library(Nebulosa)
VSMC1_density<-plot_density(TAA.combined, VSMC1_genes, reduction = "umap", joint = TRUE, combine = FALSE, pal = "magma")
VSMC2_density<-plot_density(TAA.combined, VSMC2_genes, reduction = "umap", joint = TRUE, combine = FALSE, pal = "magma")
Fibroblast1_density<-plot_density(TAA.combined, Fibroblast1_genes, reduction = "umap", joint = TRUE, combine = FALSE, pal = "magma")
Fibroblast2_density<-plot_density(TAA.combined, Fibroblast2_genes, reduction = "umap", joint = TRUE, combine = FALSE, pal = "magma")
Fibromyocyte_density<-plot_density(TAA.combined, Fibromyocyte_genes, reduction = "umap", joint = TRUE, combine = FALSE, pal = "magma")
EC1_density<-plot_density(TAA.combined, EC1_genes, reduction = "umap", joint = TRUE, combine = FALSE, pal = "magma")
EC2_density<-plot_density(TAA.combined, EC2_genes, reduction = "umap", joint = TRUE, combine = FALSE, pal = "magma")
Macrophage_density<-plot_density(TAA.combined, Macrophage_genes, reduction = "umap", joint = TRUE, combine = FALSE, pal = "magma")
NKT_density<-plot_density(TAA.combined, NKT_genes, reduction = "umap", joint = TRUE, combine = FALSE, pal = "magma")
pdf(file = "../2_Output/Supplemental_Figures/Density_plots.pdf")
VSMC1_density
VSMC2_density
Fibroblast1_density
Fibroblast2_density
EC1_density
EC2_density
Macrophage_density
NKT_density
Fibromyocyte_density
dev.off()

# Overlay these gene markers onto the UMAP to identify clusters
pdf(file = "../2_Output/CellType_FeaturePlots.pdf")
FeaturePlot(TAA.combined, reduction = "umap", label = T, features = Fibroblast1_genes)
FeaturePlot(TAA.combined, reduction = "umap", label = T, features = Fibroblast2_genes)
FeaturePlot(TAA.combined, reduction = "umap", label = T, features = EC1_genes)
FeaturePlot(TAA.combined, reduction = "umap", label = T, features = EC2_genes)
FeaturePlot(TAA.combined, reduction = "umap", label = T, features = NKT_genes)
FeaturePlot(TAA.combined, reduction = "umap", label = T, features = Macrophage_genes)
FeaturePlot(TAA.combined, reduction = "umap", label = T, features = VSMC1_genes)
FeaturePlot(TAA.combined, reduction = "umap", label = T, features = VSMC2_genes)
# FeaturePlot(TAA.combined, reduction = "umap", label = T, features = Dendritic_genes)
dev.off()
#Create a figure of cell-type specific markers overlying the UMAP
pdf(file = "../2_Output/Supplemental_Figures/CellType_Differentiation.pdf")
plot_density(TAA.combined, c("ACTA2", "FBLN1", "F13A1", "VWF", "STC1", "NFASC", "ITGAL", "ITGAX"), reduction = "umap", joint = TRUE, combine = FALSE, pal = "magma")
dev.off()
# Change the identity of the clusters to cell types
Idents(TAA.combined) <- "Celltype_annotation_v2"
# Create UMAP with annotated cell-types
UMAP_CellTypes<-DimPlot(TAA.combined,  label = T) + NoLegend()
UMAP_CellTypes
#######
umap_data <- as.data.frame(TAA.combined@reductions$umap@cell.embeddings) # Create a data.frame from the
umap_data$CellType<-TAA.combined@active.ident
cluster_centers <- aggregate(cbind(UMAP_1, UMAP_2) ~ CellType, umap_data, mean) # Label Clusters
umap_celltype <- ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, fill = CellType)) +
  # geom_point() +
  # scale_fill_brewer(palette="Set2") + 
  geom_point_trace(alpha = .25, stroke = .5, size = 1, color = "black")  +
  theme_classic() + 
  geom_text_repel(data = cluster_centers, aes(label = CellType), size = 5) +
  theme(legend.position = "none", text = element_text(size = 12))
## Fully Annotated Merged UMAP
pdf(file = "../2_Output/UMAP_Disease.Overlay.pdf", height = 5, width = 5)
umap_celltype
dev.off()

pdf(file = "../2_Output/Supplemental_Figures/UMAP_All.three_Labelled.pdf", height = 8, width = 16)
umap_disease+theme(legend.position = "topright", text = element_text(size = 14)) + 
umap_clusters+umap_celltype
dev.off()

## Use ridgeplots to identify bimodal gene marker distributions (enriched clusters)
pdf(file = "../2_Output/Celltype_RidgePlots.pdf", height = 10, width = 15)
RidgePlot(TAA.combined,features = Fibroblast1_genes, ncol = 2)
RidgePlot(TAA.combined, features = Fibroblast2_genes, ncol = 2)
RidgePlot(TAA.combined, features = EC1_genes, ncol = 2)
RidgePlot(TAA.combined, features = EC2_genes, ncol = 2)
RidgePlot(TAA.combined, features = NKT_genes, ncol = 2)
RidgePlot(TAA.combined, features = Macrophage_genes, ncol = 2)
RidgePlot(TAA.combined, features = VSMC1_genes, ncol = 2)
RidgePlot(TAA.combined, features = VSMC2_genes, ncol = 2)
RidgePlot(TAA.combined, features = Dendritic_genes, ncol = 2)
dev.off()
# Differential Expression
# Export DEGs using cell-type clusters
DEGs_CellTypes<-FindAllMarkers(TAA.combined)
write.csv(DEGs_CellTypes, "../2_Output/DEGs_Clusters.csv")
# Create a dot-bplot of the top 5 markers for each 
library(scCustomize)
paletteLength <- 100
myColor <- colorRampPalette(c("dodgerblue4", "white", "coral2"))(paletteLength)
top5_markers <- Extract_Top_Markers(marker_dataframe = DEGs_CellTypes, num_genes = 5, named_vector = FALSE,
    make_unique = TRUE)
pdf(file = "../2_Output/Dot.plot_top5markers.pdf", height = 10, width = 7)
Clustered_DotPlot(seurat_object = TAA.combined, features = top5_markers, k = 10, colors_use_exp = myColor)
dev.off()
# Heatmap of Clusters
DEGs_CellTypes %>%
    group_by(cluster) %>%
    top_n(n = 10, wt = avg_log2FC) -> top10
pdf(file = "../2_Output/Celltype_Heatmap.pdf", height = 10, width = 15)
DoHeatmap(TAA.combined, features = top10$gene, size = 3, disp.min = -2, disp.max = 2) + scale_fill_gradientn(colors = c("dodgerblue4", "white", "coral2")) + labs(title = "Heatmap of Top-10 most variable genes within each cluster")
dev.off()
# Probability distribution of specific genes across all clusters/cell types
VlnPlot(TAA.combined, features = c("VWF", "HDAC9", "ACTA2"))
#
DotPlot(TAA.combined, features = c(VSMC1_genes, VSMC2_genes, EC1_genes, EC2_genes, Fibroblast1_genes, Fibroblast2_genes, Fibromyocyte_genes, Macrophage_genes, NKT_genes)) + RotatedAxis()
saveRDS(TAA.combined, file = "TAA_FFPE_snRNA.rds")

```


# Figure 2: VSMC-Specific Analysis

### VSMC Phenotypic Comparison

```{r subpopulation, echo = FALSE, message=FALSE, warning=FALSE}
library(Seurat)
library(dplyr)
library(ggtrace)
library(ggplot2)
library(ggrepel)
# Add tracings around the points
library(ggplot2)
library(ggtrace)
library(ggthemes)
library(ggrepel)
library(dplyr)
hdat_vsmc <- subset(TAA.combined, Celltype_annotation_v2==c("VSMC1", "VSMC2"))
UMAP_VSMC<-DimPlot(hdat_vsmc,  label = T) + NoLegend()
# Figure 1A - UMAP of VSMCs only
pdf("../2_Output/Figure_2/Fig2A_UMAP_VSMC.pdf", height = 3.5, width = 3.5)
UMAP_VSMC
dev.off()

library(Nebulosa)
VSMC1_density<-plot_density(hdat_vsmc, VSMC1_genes, reduction = "umap", joint = TRUE, combine = FALSE, pal = "magma")
VSMC2_density<-plot_density(hdat_vsmc, VSMC2_genes, reduction = "umap", joint = TRUE, combine = FALSE, pal = "magma")
pdf(file = "../2_Output/Figure_2/Fig2A_VSMC_Density_plots.pdf", height = 3.5, width = 3.5)
VSMC1_density
VSMC2_density
dev.off()

# Visualization
p1 <- DimPlot(hdat_vsmc, reduction = "umap", group.by = "Disease")
p2 <- DimPlot(hdat_vsmc, reduction = "umap", label = TRUE, repel = TRUE)
p1 + p2

umap_data <- as.data.frame(hdat_vsmc@reductions$umap@cell.embeddings) # Create a data.frame from the UMAP (to be used in ggplot)
# Disease Clustering
umap_data$Disease <- hdat_vsmc$Disease
umap_data$Disease <- factor(umap_data$Disease, levels = c("Control", "Dissection"))
p1 <- ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, fill = Disease)) +
  geom_point_trace(alpha = .25, stroke = .5, size = 1, color = "black")  +
  theme_classic() +
  theme(text = element_text(size = 14))
p1
umap_TAA<-umap_data %>% filter(Disease=="Control")
p_TAA <- ggplot(umap_TAA, aes(x = UMAP_1, y = UMAP_2, fill = Disease)) +
  geom_point_trace(alpha = .25, stroke = .5, size = 1, color = "black")  +
  theme_classic() +
  theme(text = element_text(size = 14))
p_TAA
umap_ACTA<-umap_data %>% filter(Disease=="Dissection")
p_ACTA <- ggplot(umap_ACTA, aes(x = UMAP_1, y = UMAP_2, fill = Disease)) +
  geom_point_trace(alpha = .25, stroke = .5, size = 1, color = "black")  +
  theme_classic() +
  theme(text = element_text(size = 14))
p1 + p_TAA + p_ACTA

# Create the UMAP of cell-type specific clusters
UMAP_CellTypes<-DimPlot(hdat_vsmc, label = T) + NoLegend()
UMAP_CellTypes
umap_data <- as.data.frame(hdat_vsmc@reductions$umap@cell.embeddings) # Create a data.frame from the
umap_data$Cluster<-hdat_vsmc@active.ident
cluster_centers <- aggregate(cbind(UMAP_1, UMAP_2) ~ Cluster, umap_data, mean) # Label Clusters
p3 <- ggplot(umap_data, aes(x = UMAP_1, y = UMAP_2, fill = Cluster)) +
  # geom_point() +
  # scale_fill_brewer(palette="Set2") + 
  geom_point_trace(alpha = .25, stroke = .5, size = 1, color = "black")  +
  theme_classic() + 
  geom_text_repel(data = cluster_centers, aes(label = Cluster), size = 6) +
  theme(legend.position = "none", text = element_text(size = 14))
p3
pdf(file = "../2_Output/Figure_2/VSMC_UMAP_CellTypes.pdf", width = 11.5, height = 5, bg = "transparent")
p1+theme(legend.position = "top", text = element_text(size = 14)) + p3
dev.off()

# Differential Expression between VSMCs
VSMC1.markers <- FindMarkers(hdat_vsmc, ident.1 = "VSMC1", min.pct = 0.25, logfc.threshold = 0.25)
VSMC2.markers <- FindMarkers(hdat_vsmc, ident.1 = "VSMC2", min.pct = 0.25, logfc.threshold = 0.25)

#Volcano Plot
# Load packages
library(dplyr)
library(ggplot2)
library(ggrepel)
library(openxlsx)
# Read data from the web
options(ggrepel.max.overlaps = Inf)
results = mutate(VSMC2.markers, minuslogpvalue = -log(p_val), log2FC=avg_log2FC)
results<-results %>% filter(p_val!=0)
results$gene_name<-rownames(results)
results <- results %>% 
  mutate(., sig=ifelse(p_val<0.05 & log2FC>.5, 
                       "P < 0.05 and Fold-Change > 0.5", 
                       ifelse(p_val<0.05 & log2FC< 0-0.5,
                              "P < 0.05 and Fold-Change < -0.5", 
                              "Not Sig")
                       )
         )
results$sig<-factor(results$sig, 
levels = c("P < 0.05 and Fold-Change < -0.5",
  "Not Sig",
  "P < 0.05 and Fold-Change > 0.5")
  )
max(results$minuslogpvalue, na.rm = TRUE)
max(results$log2FC, na.rm = TRUE)
min(results$log2FC, na.rm = TRUE)
p = ggplot(results, aes(log2FC, minuslogpvalue)) + 
  theme_classic() +
  theme(axis.line = element_blank(),
        axis.ticks = element_blank()
  ) +
  geom_point(aes(fill=sig, size = minuslogpvalue),
             colour="black",
             shape=21,
             stroke = 0,
             alpha = .9) +
  geom_vline(xintercept=0.5, size=.5, linetype="dashed") +
  geom_vline(xintercept=-0.5, size=0.5, linetype="dashed") +
  geom_hline(yintercept=0-log(0.05), size=.5, linetype="dashed") +
  labs(x=expression(Log[2](Fold-Change)), y=expression(-Log[10](P-value))) + 
  xlim(min(results$log2FC, na.rm = TRUE),max(results$log2FC, na.rm = TRUE)) + 
  ylim(0, max(results$minuslogpvalue, na.rm = TRUE)) + geom_hline(yintercept = 0, size = 1) + 
  geom_vline(xintercept=0, size=1) +
  scale_fill_manual(values=c("darkcyan", "darkgray", "coral2")) +
  scale_size_continuous(range = c(.1, 3))

  p+
  geom_text_repel(data=top_n(filter(results, log2FC< -0.5), 10, minuslogpvalue), aes(label=gene_name)) +
  geom_text_repel(data=top_n(filter(results, log2FC>0.5), 10, minuslogpvalue), aes(label=gene_name)) +
  theme(text = element_text(size=20))
##
pdf(file = paste0("../2_Output/Figure_2/VSMC_1.vs.2_VolcanoPlot.pdf"), height = 5, width = 5)
  p+
  geom_text_repel(data=top_n(filter(results, log2FC< -0.5), 10, minuslogpvalue), aes(label=gene_name)) +
  geom_text_repel(data=top_n(filter(results, log2FC>0.5), 10, minuslogpvalue), aes(label=gene_name)) +
    theme(text = element_text(size=14), legend.position="none")
dev.off()
# Create a file of differentially expressed genes between VSMC1 and VSMC2
openxlsx::write.xlsx(VSMC1.markers, file = "../2_Output/Figure_2/VSMC_1v2_DEGs.xlsx", rowNames=T)

VSMC_type_DEs <- results$gene_name[1:100]
##
plots_VSMC2 <- VlnPlot(hdat_vsmc, features = c("PALLD", "MAP2", "PRKG1"), group.by = "Celltype_annotation_v2",
    pt.size = 0) & theme(legend.position = 'none', axis.title.x = element_blank())
plots_VSMC1 <- VlnPlot(hdat_vsmc, features = c("B2M", "TNFAIP2", "STAB1"), group.by = "Celltype_annotation_v2",
    pt.size = 0) & theme(legend.position = 'none', axis.title.x = element_blank())

pdf(file = "../2_Output/Figure_2/VSMC2_Top.Violins.pdf", height = 3, width = 6)
plots_VSMC2
dev.off()
pdf(file = "../2_Output/Figure_2/VSMC1_Top.Violins.pdf", height = 3, width = 6)
plots_VSMC1
dev.off()

##################
DoHeatmap(hdat_vsmc, features = VSMC_type_DEs, group.by = "Celltype_annotation_v2",size = 3, disp.min = -2, disp.max = 2) + scale_fill_gradient2(low = "dodgerblue4", mid = "white", high = "coral2",midpoint = 0)
##################
pdf(file = "../2_Output/Figure_2/DotPlot_VSMC_1.v.2.pdf", height = 10, width = 5)
Clustered_DotPlot(seurat_object = hdat_vsmc, features = VSMC_type_DEs, group.by = "Celltype_annotation_v2", x_lab_rotate=F, k = 2)
dev.off()
```

### VSMC Trajectory Analysis (Monocle 3)

```{r monocle3_Merged, echo = FALSE, message=FALSE, warning=FALSE}
library(monocle3)
library(Seurat)
library(SeuratWrappers)
library(dplyr)
library(ggtrace)
Merged_scaled<-hdat_vsmc
Merged_scaled@active.assay = "RNA"
cds<-SeuratWrappers::as.cell_data_set(Merged_scaled)
cds <- preprocess_cds(cds, num_dim = 100)
cds <- estimate_size_factors(cds)
# Include gene names (not done by default by the seurat conversion)
rowData(cds)$gene_name <- rownames(cds)
rowData(cds)$gene_short_name <- rowData(cds)$gene_name
#  Run Monocle
cds <- cluster_cells(cds) # This step creates "partitions" that are used in the trajectory inference
plot_cells(cds, show_trajectory_graph = FALSE, color_cells_by = "partition") # this shows the partitions overlain on the UMAP
cds <- learn_graph(cds, use_partition = TRUE) # creating trajectory inference within each partition
# cds <- order_cells(cds) # Used when setting the nodes; if already known, use the next line
root_cell <- "TACTCGCTCCCTGACT-1-1" # names(which(cds@principal_graph_aux$UMAP$pseudotime==0))
cds<-order_cells(cds, root_cells = root_cell)

# Plot the pseudotime on UMAP
plot_cells(cds, 
           color_cells_by = "pseudotime",
           label_branch_points = FALSE,
           label_leaves = FALSE)
pdf(file = "../2_Output/Figure_2/VSMC_UMAP_Trajectory_Partition.pdf", height = 3, width = 3)
plot_cells(cds,
           color_cells_by = "partition",
           graph_label_size = 1,
           cell_size = .5,
           label_roots = F,
           label_branch_points = FALSE,
           label_leaves = FALSE)
dev.off()
pdf(file = "../2_Output/Figure_2/VSMC_UMAP_Trajectory_Pseudotime.pdf", height = 3, width = 4)
plot_cells(cds,
           color_cells_by = "pseudotime",
           graph_label_size = 1,
           cell_size = .7,
           label_branch_points = FALSE,
           label_leaves = F)
dev.off()

# Identify pseudotime
modulated_genes <- graph_test(cds, neighbor_graph="principal_graph", cores=8) # Identify differentially-expressed genes with pseudotime
modulated_genes <- na.omit(modulated_genes) # remove NA's
modulated_genes <- modulated_genes[modulated_genes$q_value < 0.05 & modulated_genes$status =="OK", ] # filter cds results down
modulated_genes <- modulated_genes[order(-modulated_genes$morans_test_statistic), ] # order by moran's test
#### Create a heatmap of genes with similar pseudotime kinetics
genes <- row.names(subset(modulated_genes, q_value < 0.05 & morans_I > 0.2))
openxlsx::write.xlsx(modulated_genes, "../2_Output/Figure_2/Pseudotime_DEGs.xlsx")

library(ComplexHeatmap)
library(ggplot2)
library(dplyr)
library(RColorBrewer)
library(circlize)
library(monocle3)
pt.matrix <- exprs(cds)[match(genes,rownames(rowData(cds))),order(pseudotime(cds))]
#Can also use "normalized_counts" instead of "exprs" to use various normalization methods, for example:
#normalized_counts(cds, norm_method = "log")
pt.matrix <- t(apply(pt.matrix,1,function(x){smooth.spline(x,df=3)$y})) # Create a spline that smooths the pseudotime-based expression along the first 3 degrees of freedom.
pt.matrix <- t(apply(pt.matrix,1,function(x){(x-mean(x))/sd(x)}))
rownames(pt.matrix) <- genes;

#Ward.D2 Hierarchical Clustering
hthc <- Heatmap(
  pt.matrix,
  name                         = "z-score",
  col                          = colorRamp2(seq(from=-2,to=2,length=11),rev(brewer.pal(11, "Spectral"))),
  show_row_names               = TRUE,
  show_column_names            = FALSE,
  row_names_gp                 = gpar(fontsize = 6),
  clustering_method_rows = "ward.D2",
  clustering_method_columns = "ward.D2",
  row_title_rot                = 0,
  cluster_rows                 = TRUE,
  cluster_row_slices           = FALSE,
  cluster_columns              = FALSE)
pdf(file = "../2_Output/Figure_2/VSMC_Pseudotime.Heatmap.pdf", height = 4, width = 3)
print(hthc)
dev.off()

# Examine specific genes within the module(s) of interest (appears to emrich to metabolic genes)
library("viridis")
library(ggplot2)
GENES <- c("AEBP1", "LAMA2", "HDAC9", "ACTA2", "EXOC4", "PDE3A") #Genes selected based on visible trends
p1 <- plot_cells(cds, 
           genes = GENES[1:2],
           label_cell_groups = F,
           label_roots = F,
           label_branch_points = F,
           label_leaves = F,
           show_trajectory_graph = F,
           min_expr = 1,
           alpha = 0.8,
           trajectory_graph_color = "grey28",
           ) +
  # geom_point_trace(alpha = .25, stroke = .5, size = 1, color = "black")  +
  theme(
  axis.text = element_text(size = 6),    # Adjust the size as needed
  axis.title = element_text(size = 8),  # Adjust the size as needed
  legend.text = element_text(size = 4),  # Adjust the size as needed
  legend.title = element_text(size = 6),
  legend.key.size = unit(2, 'mm')) +
  scale_colour_viridis_c(option = "inferno")
p2 <- plot_cells(cds, 
           genes = GENES[3:4],
           label_cell_groups = F,
           label_roots = F,
           label_branch_points = F,
           label_leaves = F,
           show_trajectory_graph = F,
           min_expr = 1,
           alpha = 0.8,
           trajectory_graph_color = "grey28",
           ) +
  # geom_point_trace(alpha = .25, stroke = .5, size = 1, color = "black")  +
  theme(
  axis.text = element_text(size = 6),    # Adjust the size as needed
  axis.title = element_text(size = 8),  # Adjust the size as needed
  legend.text = element_text(size = 4),  # Adjust the size as needed
  legend.title = element_text(size = 6),
  legend.key.size = unit(2, 'mm')) +
  scale_colour_viridis_c(option = "inferno")
p3 <- plot_cells(cds, 
           genes = GENES[5:6],
           label_cell_groups = F,
           label_roots = F,
           label_branch_points = F,
           label_leaves = F,
           show_trajectory_graph = F,
           min_expr = 1,
           alpha = 0.8,
           trajectory_graph_color = "grey28",
           ) +
  # geom_point_trace(alpha = .25, stroke = .5, size = 1, color = "black")  +
  theme(
  axis.text = element_text(size = 6),    # Adjust the size as needed
  axis.title = element_text(size = 8),  # Adjust the size as needed
  legend.text = element_text(size = 4),  # Adjust the size as needed
  legend.title = element_text(size = 6),
  legend.key.size = unit(2, 'mm')) +
  scale_colour_viridis_c(option = "inferno")
pdf(file="../2_Output/Figure_2/UMAP_Pseudotime_Genes.pdf", height = 4, width = 3)
ggarrange(p1, p2,p3, ncol=1, nrow=3, common.legend = TRUE, legend="right")
dev.off()
p1+p2+p3
# Plot genes according to "pseudotime"
lineage_cds <- cds[rowData(cds)$gene_short_name %in% GENES, ] #colData(cds)$cell_type %in% c("VSMC")
lineage_cds<-order_cells(lineage_cds, root_cells = root_cell)
pdf(file="../2_Output/Figure_2/Pseudotime_curves.pdf", height = 7, width = 3)
monocle3::plot_genes_in_pseudotime(lineage_cds,
                         color_cells_by="ident",
                         min_expr=0)
dev.off()

```

######################################
### VSMC Differential Expression (Dissection vs. Control)
######################################

```{r DEGs_Disease, echo = FALSE, message=FALSE, warning=FALSE}
library(ggpubr)
hdat_vsmc$VSMC_phenotype <- Idents(hdat_vsmc)
Idents(hdat_vsmc) <- "Disease"
de_VSMC <- FindMarkers(hdat_vsmc, ident.1 = "Dissection", ident.2 = "Control", verbose = FALSE, logfc.threshold = 0.25, min.pct = 0.25)
# Create a file of differentially expressed genes between VSMC1 and VSMC2
openxlsx::write.xlsx(de_VSMC, file = "../2_Output/Figure_2/VSMC_DEGs_Disease.xlsx", rowNames=T)
###
library(ggpubr)
plots_Disease_UP <- VlnPlot(hdat_vsmc, features = c("MACROD2", "CDK17", "FOS"), group.by = "Disease",
    pt.size = 0) & theme(axis.title.x = element_blank()) & scale_fill_manual(values = c("dodgerblue4", "goldenrod2"))
plots_Disease_DOWN <- VlnPlot(hdat_vsmc, features = c("PTPRQ", "HLA-B", "STAB1"), group.by = "Disease",
    pt.size = 0) & theme(axis.title.x = element_blank()) & scale_fill_manual(values = c("dodgerblue4", "goldenrod2"))
# ggarrange(plots_Disease_UP, plots_Disease_DOWN, ncol=1, nrow=2, common.legend = TRUE, legend="right")

pdf(file = "../2_Output/Figure_2/Dissection_up_Top.Violins.pdf", height = 3, width = 6)
plots_Disease_UP
dev.off()
pdf(file = "../2_Output/Figure_2/Dissection_down_Top.Violins.pdf", height = 3, width = 6)
plots_Disease_DOWN
dev.off()
##################
DoHeatmap(hdat_vsmc, features = rownames(de_VSMC), size = 3, disp.min = -2, disp.max = 2) + scale_fill_gradient2(low = "dodgerblue4", mid = "white", high = "coral2",midpoint = 0)
##################
pdf(file = "../2_Output/Figure_2/Dot.plot_top5markers.pdf", height = 10, width = 7)
Clustered_DotPlot(seurat_object = hdat_vsmc, features = rownames(de_VSMC), x_lab_rotate=F, k = 10)
dev.off()

#Volcano Plot
# Load packages
library(dplyr)
library(ggplot2)
library(ggrepel)
library(openxlsx)
# Read data from the web
options(ggrepel.max.overlaps = Inf)
results = mutate(de_VSMC, minuslogpvalue = -log(p_val), log2FC=avg_log2FC)
results<-results %>% filter(p_val!=0)
results$gene_name<-rownames(results)
results <- results %>% 
  mutate(., sig=ifelse(p_val<0.05 & log2FC>.5, 
                       "P < 0.05 and Fold-Change > 0.5", 
                       ifelse(p_val<0.05 & log2FC< 0-0.5,
                              "P < 0.05 and Fold-Change < -0.5", 
                              "Not Sig")
                       )
         )
results$sig<-factor(results$sig, 
levels = c("P < 0.05 and Fold-Change < -0.5",
  "Not Sig",
  "P < 0.05 and Fold-Change > 0.5")
  )
max(results$minuslogpvalue, na.rm = TRUE)
max(results$log2FC, na.rm = TRUE)
min(results$log2FC, na.rm = TRUE)
p = ggplot(results, aes(log2FC, minuslogpvalue)) + 
  theme_classic() +
  theme(axis.line = element_blank(),
        axis.ticks = element_blank()
  ) +
  geom_point(aes(fill=sig, size = minuslogpvalue),
             colour="black",
             shape=21,
             stroke = 0,
             alpha = 0.9) +
  geom_vline(xintercept=0.5, size=.5, linetype="dashed") +
  geom_vline(xintercept=-0.5, size=0.5, linetype="dashed") +
  geom_hline(yintercept=0-log(0.05), size=.5, linetype="dashed") +
  labs(x=expression(Log[2](Fold-Change)), y=expression(-Log[10](P-value))) + 
  xlim(min(results$log2FC, na.rm = TRUE),max(results$log2FC, na.rm = TRUE)) + 
  ylim(-0, max(results$minuslogpvalue, na.rm = TRUE)) + geom_hline(yintercept = 0, size = 1) + 
  geom_vline(xintercept=0, size=1) +
  scale_fill_manual(values=c("dodgerblue4", "darkgray", "goldenrod2")) +
  scale_size_continuous(range = c(.1, 3))

  p+
  geom_text_repel(data=top_n(filter(results, log2FC< -0.5), 20, minuslogpvalue), aes(label=gene_name)) +
  geom_text_repel(data=top_n(filter(results, log2FC>0.5), 20, minuslogpvalue), aes(label=gene_name)) +
  theme(text = element_text(size=20))
##
pdf(file = paste0("../2_Output/Figure_2/VSMC_Dissection.vs.Control_VolcanoPlot.pdf"), height = 5, width = 5)
  p+
  geom_text_repel(data=top_n(filter(results, log2FC< -0.5), 20, minuslogpvalue), aes(label=gene_name)) +
  geom_text_repel(data=top_n(filter(results, log2FC>0.5), 20, minuslogpvalue), aes(label=gene_name)) +
    theme(text = element_text(size=14), legend.position="none")
dev.off()
# 
# ##################
# DoHeatmap(hdat_vsmc, features = VSMC_type_DEs, group.by = "VSMC_phenotype",size = 3, disp.min = -2, disp.max = 2) + scale_fill_gradientn(colors = c("dodgerblue4", "white", "coral2"))
##################
pdf(file = "../2_Output/Figure_2/DotPlot_VSMC_1.v.2.pdf", height = 10, width = 5)
Clustered_DotPlot(seurat_object = hdat_vsmc, features = VSMC_type_DEs, group.by = "VSMC_phenotype", x_lab_rotate=F, k = 2)
dev.off()
```

# Supplemental Table: R Session Information
All packages and setting are acquired using the following command:

```{r settings}
sinfo<-devtools::session_info()
sinfo$platform
sinfo$packages %>% kable( 
                         align="c", 
                         longtable=T, 
                         booktabs=T,
                         caption="Packages and Required Dependencies") %>% 
    kable_styling(latex_options=c("striped", "repeat_header", "condensed"))
```
